derive-input/src/internal/dsl/method.rs

```diff
@@ -1360,12 +1390,34 @@ pub(in crate::internal) fn for_method(
                                 });
                             };
 
+                            let hook_call = if spacetimedsl_table.hook_on_delete {
+                                let hook_fn_name = format_ident!("on_{}_delete", singular_table_name);
+                                quote! {
+                                    // Collect rows before deletion to call hooks on them
+                                    // We need to collect because filter_map would consume the rows,
+                                    // but we need to keep them to call hooks before actual deletion
+                                    let rows_to_delete: Vec<_> = primary_key_values_of_rows_to_delete.iter().filter_map(|pk| {
+                                        self.ctx().db().#singular_table_name().#primary_key_column_name().find(*pk)
+                                    }).collect();
```

Comment:
do we really need to fetch all of them? We should give the IDs instead to the hook and devs should decide whether they want to fetch the data or just do something else with it.